--- a/libs/libkexiv2/libkexiv2/subjectwidget.cpp
+++ b/libs/libkexiv2/libkexiv2/subjectwidget.cpp
@@ -595,4 +595,10 @@
     return newSubjects;
 }
 
+//KDE5: Remove me
+void SubjectWidget::setSubjectList(const QStringList& list)
+{
+    setSubjectsList(list);
+}
+
 }  // namespace KExiv2Iface
--- a/libs/libkexiv2/libkexiv2/subjectwidget.h
+++ b/libs/libkexiv2/libkexiv2/subjectwidget.h
@@ -78,6 +78,10 @@
     void setSubjectsList(const QStringList& list);
     QStringList subjectsList() const;
 
+    /** @deprecated Provided to keep binary compatibility with
+     * older versions. Use setSubjectsList() instead. */
+    KDE_DEPRECATED void setSubjectList(const QStringList& list);
+
 Q_SIGNALS:
 
     void signalModified();
