Index: kdegraphics/okular/core/area.cpp
===================================================================
--- kdegraphics/okular/core/area.cpp	(revision 837842)
+++ kdegraphics/okular/core/area.cpp	(revision 837843)
@@ -42,7 +42,11 @@
 
 void NormalizedPoint::transform( const QMatrix &matrix )
 {
-    matrix.map( x, y, &x, &y );
+    qreal tmp_x = (qreal)x;
+    qreal tmp_y = (qreal)y;
+    matrix.map( tmp_x, tmp_y, &tmp_x, &tmp_y );
+    x = tmp_x;
+    y = tmp_y;
 }
 
 QDebug operator<<( QDebug str, const Okular::NormalizedPoint& p )
