Index: kdegraphics/okular/generators/xps/generator_xps.cpp
===================================================================
--- kdegraphics/okular/generators/xps/generator_xps.cpp	(revision 839972)
+++ kdegraphics/okular/generators/xps/generator_xps.cpp	(revision 839973)
@@ -981,7 +981,7 @@
         QVector<qreal> dashPattern( pieces.count() );
         bool ok = false;
         for ( int i = 0; i < pieces.count(); ++i ) {
-            double value = pieces.at( i ).toInt( &ok );
+            qreal value = pieces.at( i ).toInt( &ok );
             if ( ok ) {
                 dashPattern[i] = value;
             } else {
Index: kdegraphics/okular/ui/pageviewutils.cpp
===================================================================
--- kdegraphics/okular/ui/pageviewutils.cpp	(revision 839972)
+++ kdegraphics/okular/ui/pageviewutils.cpp	(revision 839973)
@@ -102,12 +102,12 @@
     return m_zoomFactor;
 }
 
-qreal PageViewItem::absToPageX( qreal absX ) const
+double PageViewItem::absToPageX( double absX ) const
 {
     return ( absX - m_uncroppedGeometry.left() ) / m_uncroppedGeometry.width();
 }
 
-qreal PageViewItem::absToPageY( qreal absY ) const
+double PageViewItem::absToPageY( double absY ) const
 {
     return ( absY - m_uncroppedGeometry.top() ) / m_uncroppedGeometry.height();
 }
